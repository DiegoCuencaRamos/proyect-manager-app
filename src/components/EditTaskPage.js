import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory, useParams } from 'react-router-dom'
import Title from './Title'
import { startEditTask } from '../actions/task'
import ItemForm from './Form'

const EditTaskPage = () => {
    const tasks = useSelector(state => state.tasks)
    const proyectId = useSelector(state => state.ids.proyectId)
    const { taskId } = useParams()
    const task = tasks.find(task => task.id === taskId)
    const history = useHistory()

    const dispatch = useDispatch()

    const onParentFormSubmit = (updates) => {
        dispatch(startEditTask(taskId, updates))
        history.push(`/proyect/${proyectId}`)
    }

    return(
        <section>
            <Title
                title={'Edit task'}
                description={'Edit your current task'}
            />

            <ItemForm
                proyectId={proyectId}
                item={task}
                onParentFormSubmit={onParentFormSubmit}
            />
        </section>
    )
}

export default EditTaskPage