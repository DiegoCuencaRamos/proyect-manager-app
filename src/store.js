import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store