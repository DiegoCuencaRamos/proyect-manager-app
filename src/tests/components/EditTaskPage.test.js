import React from 'react'
import { mount, render } from 'enzyme'
import { Provider } from 'react-redux'
import store from '../../store'
import EditTaskPage from '../../components/EditTaskPage'
import { MemoryRouter } from 'react-router'

const getWrapper = () => mount(
    <MemoryRouter>
        <Provider store={store}>
            <EditTaskPage />
        </Provider>
    </MemoryRouter>
)

test('Should render edit page correctly', () => {
    const wrapper = render(
        <MemoryRouter>
            <Provider store={store}>
                <EditTaskPage />
            </Provider>
        </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
})

// Falta testear la parte del dispatch