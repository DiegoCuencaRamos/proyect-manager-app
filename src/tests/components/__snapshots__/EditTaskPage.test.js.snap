// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render edit page correctly 1`] = `
<section>
  <div
    class="page-header"
  >
    <div
      class="container page-header__wrapper"
    >
      <div>
        <h1
          class="page-header__title"
        >
          Edit task
        </h1>
        <p
          class="page-header__text"
        >
          Edit your current task
        </p>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <form
      class="form"
    >
      <div
        class="form__wrapper"
      >
        <input
          class="form__item--name"
          placeholder="Name"
          type="text"
          value=""
        />
        <select
          class="form__item--status"
          style="flex-basis:100%;margin-right:0"
        >
          <option
            selected=""
            value=""
          >
            Select status
          </option>
          <option
            value="todo"
          >
            To do
          </option>
          <option
            value="doing"
          >
            Doing
          </option>
          <option
            value="done"
          >
            Done
          </option>
        </select>
        <div
          class="DateRangePicker DateRangePicker_1"
        >
          <div>
            <div
              class="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2 DateRangePickerInput__showClearDates DateRangePickerInput__showClearDates_3"
            >
              <div
                class="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-0"
                  aria-label="Start Date"
                  autocomplete="off"
                  class="DateInput_input DateInput_input_1"
                  id="0"
                  name="0"
                  placeholder="Start Date"
                  type="text"
                  value="01/01/1970"
                />
                <p
                  class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-0"
                >
                  Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
              <div
                aria-hidden="true"
                class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                role="presentation"
              >
                <svg
                  class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                  focusable="false"
                  viewBox="0 0 1000 1000"
                >
                  <path
                    d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                  />
                </svg>
              </div>
              <div
                class="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-0"
                  aria-label="End Date"
                  autocomplete="off"
                  class="DateInput_input DateInput_input_1"
                  id="0"
                  name="0"
                  placeholder="End Date"
                  type="text"
                  value="01/01/1970"
                />
                <p
                  class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-0"
                >
                  Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
              <button
                aria-label="Clear Dates"
                class="DateRangePickerInput_clearDates DateRangePickerInput_clearDates_1 DateRangePickerInput_clearDates_default DateRangePickerInput_clearDates_default_2"
                type="button"
              >
                <svg
                  class="DateRangePickerInput_clearDates_svg DateRangePickerInput_clearDates_svg_1"
                  focusable="false"
                  viewBox="0 0 12 12"
                >
                  <path
                    d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <label
          class="form__item--color"
        >
          Task color:
          <input
            type="color"
            value=""
          />
        </label>
        <textarea
          class="form__item--description"
          placeholder="Description"
        />
        <input
          class="form__item--submit button"
          type="submit"
          value="Save task"
        />
      </div>
    </form>
  </div>
</section>
`;
